.github/workflows/ci.yml

```diff
@@ -151,47 +151,36 @@ jobs:
       run: |
         codecov -n "GitHub Actions - ${{ matrix.task.name}} - ${{ matrix.os.name }} ${{ matrix.python.name }}"
 
-  check:
-    name: ${{ matrix.task.name}} - ${{ matrix.python.name }}
+  lint:
+    name: Linters
     runs-on: ubuntu-latest
-    needs:
-      - build
-    strategy:
-      fail-fast: false
-      matrix:
-        python:
-          # Using second most recent minor release for whatever little
-          # increase in stability over using the latest minor.
-          - name: CPython 3.9
-            tox: py39
-            action: 3.9
-        task:
-          - name: Check Newsfragment
-            tox: check-newsfragment
+    env:
+      TOX_PARALLEL_NO_SPINNER: 1
 
     steps:
     - uses: actions/checkout@v3
       with:
         fetch-depth: 0
 
-    - name: Download package files
-      uses: actions/download-artifact@v3
-      with:
-        name: dist
-        path: dist/
-
-    - name: Set up ${{ matrix.python.name }}
+    - name: Set up python
       uses: actions/setup-python@v3
       with:
-        python-version: ${{ matrix.python.action }}
+        python-version: "3.10"
 
     - name: Install dependencies
       run: python -m pip install --upgrade pip tox
 
     - uses: twisted/python-info-action@v1
 
     - name: Tox
-      run: tox -c tox.ini -e ${{ matrix.task.tox }}
+      run: tox -c tox.ini --parallel -e pre-commit
+
+    - name: Tox
+      run: tox -c tox.ini --parallel -e check-newsfragment
+
+    - name: Tox
+      run: tox -c tox.ini --parallel -e check-manifest
```

Comment:
Use the relevant names for the step name.  Why parallel for single environments?  How about `if: always()` so you get all results in a single run rather than having to iterate through multiple failures via multiple commits and ci runs.